int8 x
int8 y
string emotion
